tests/creation.test.ts

```diff
@@ -0,0 +1,151 @@
+
+    beforeEach(async () => {
+      ;({ safeSingletonAddress, proxyFactoryAddress } = await setupTests())
+    })
+
+    it('should fail if the signer is not connected to a provider', async () => {
+      // Used to mock a signer without provider
+      const mockedSigner = {}
+      const ethersSafeFactory = new EthersSafeFactory(mockedSigner as Signer, {
+        safeSingletonAddress,
+        proxyFactoryAddress
+      })
+      await chai
+        .expect(
+          ethersSafeFactory.createSafe({
+            owners,
+            threshold: -1
+          })
+        )
+        .rejectedWith('Signer must be connected to a provider')
+    })
+
+    it('should fail if the proxyFactoryAddress is not valid', async () => {
+      const ethersSafeFactory = new EthersSafeFactory(user1, {
+        safeSingletonAddress,
+        proxyFactoryAddress: '0x0DA0C3e52C977Ed3cBc641fF02DD271c3ED55aFe'
+      })
+      await chai
+        .expect(
+          ethersSafeFactory.createSafe({
+            owners,
+            threshold: -1
+          })
+        )
+        .rejectedWith('ProxyFactory contract is not deployed in the current network')
+    })
+
+    it('should fail if the safeSingletonAddress is not valid', async () => {
+      const ethersSafeFactory = new EthersSafeFactory(user1, {
+        safeSingletonAddress: '0x0DA0C3e52C977Ed3cBc641fF02DD271c3ED55aFe',
+        proxyFactoryAddress
+      })
+      await chai
+        .expect(
+          ethersSafeFactory.createSafe({
+            owners,
+            threshold: -1
+          })
+        )
+        .rejectedWith('SafeSingleton contract is not deployed in the current network')
+    })
+
+    it('should fail if the threshold is less than or equal to zero', async () => {
+      const ethersSafeFactory = new EthersSafeFactory(user1, {
+        safeSingletonAddress,
+        proxyFactoryAddress
+      })
+      await chai
+        .expect(
+          ethersSafeFactory.createSafe({
+            owners,
+            threshold: -1
+          })
+        )
+        .rejectedWith('Invalid threshold: it must be greater than or equal to')
+    })
+
+    it('should fail if the threshold is greater than the owners length', async () => {
+      const ethersSafeFactory = new EthersSafeFactory(user1, {
+        safeSingletonAddress,
+        proxyFactoryAddress
+      })
+      await chai
+        .expect(
+          ethersSafeFactory.createSafe({
+            owners,
+            threshold: 4
+          })
+        )
+        .rejectedWith('Invalid threshold: it must be lower than or equal to owners length')
+    })
+
+    it('should successfully create a safeSDK instance if the threshold is set properly', async () => {
+      const ethersSafeFactory = new EthersSafeFactory(user1, {
+        safeSingletonAddress,
+        proxyFactoryAddress
+      })
+      const safeSdk = await ethersSafeFactory.createSafe({
+        owners,
+        threshold: 2
+      })
+      chai.expect(safeSdk).not.to.be.undefined
```

Comment:
We can check for example
```suggestion
      chai.expect(await safeSdk.getOwners()).to.be.eq(owners)
      chai.expect(await safeSdk.getThreshold()).to.be.eq(2)
```